gpu/api.go

```diff
@@ -189,12 +200,18 @@ func WithComputePow(difficulty []byte) OptionFunc {
 	}
 }
 
+func WithScryptParams(n, r, p uint32) OptionFunc {
+	return func(opts *option) error {
+		opts.n = n
+		opts.r = r
+		opts.p = p
+		return nil
+	}
+}
+
 // ScryptPositions computes the scrypt output for the given options.
 func ScryptPositions(opts ...OptionFunc) (*ScryptPositionsResult, error) {
 	options := &option{
-		n:             512,
-		r:             1,
-		p:             1,
```

Comment:
I like the `WithScryptParams` functional option, but I would still leave default values for these parameters.

It will always be `WithScryptParams(8192, 1,1)` outside of tests. I think we keep default values (and only change `n` from 512 to 8192)

Reply:
I think that the C-library wrapper shouldn't define the defaults and require params to be explicitly set.